File: reddit_challenges/277-20170519/277.py
import sys

def gcd(a, b):

    # find out which is larger and set the range top var
	if a-b < 0:
		x = a
	else:
		x = b

    # modular arithmetic again! I have an unreasonable attraction here...
	for i in range(x, 2, -1):
		if a%i == 0 and b%i == 0:
			return i
	return 1

def main():
    a = int(sys.argv[1])    # have to convert to an int, just like C
    b = int(sys.argv[2])
    c = gcd(a,b)            # store results in c
    print("gcd={}  num={}  den={}".format(c, a/c, b/c))

# account for the main() implementation
if __name__ == "__main__":
    main()

File: reddit_challenges/277-20170519/277.c
#include <stdio.h>
#include <stdlib.h>

int gcd(int, int);

int main(int argc, char **argv) {
  int num = atoi(argv[1]);  //casting doesn't work here, so use atoi
  int den = atoi(argv[2]);
  int res = gcd(num, den);  //store the results

  printf("%d %d\n", num/res, den/res);

  return 0;
}

int gcd(int a, int b) {
  //check to see which of the params are smaller
  if((b-a) < 0)
    return gcd(b, a);

  //the meat of gcd, modular arithmetic!
  for(int i=a; i>1; i--) {
    if(a%i == 0 && b%i == 0)
      return i;
  }
  return 1;
}
